### ISSUE TITLE
How to use as dependency with continuous integration environment

### ISSUE THREAD

**Issue by ejunker** [daquinoaldo/https-localhost#9]
I have a project that has added https-localhost as a dependency. We have a Jenkins continuous integration server that checks out our code from git and then installs the npm modules so that it can build our app. After the npm modules are installed, the postinstall script for https-localhost runs and it generates the certificates but then get prompted to enter a password. I was wondering how other people handle this situation. I know I could pass the `--ignore-scripts` flag to npm but that ignores **all** post-install scripts and I cannot do that.

```
> https-localhost@3.1.2 postinstall /opt/node/myapp/node_modules/https-localhost
> node cert/generate.js

Generating certificates...
Downloading the mkcert executable...
Running mkcert to generate certificates...
[sudo] password for appuser:
```

You could possibly check for an environment variable and then if present skip the postinstall.

```
https-localhost_skip_postintall="yes" npm install
```
Then `cert/generate.js` would have to updated to check for the `https-localhost_skip_postinstall` environment variable and not generate the certificates.

**Comment by daquinoaldo** [comment]
Hi! Thank you for opening an issue!

https-localhost was designed for local use, on your PC, and not for production or for CI.
Anyway, if you could explain better the goal you want to reach with https-localhost on Jenkins I'll be glad to help you to find a work-around, also by adding some checks in the script.

First please note the issues #8 and #10. I'm planning to implement them this week.
Now the script act this way: at install time the post install script generates certificates in ./certs.
Then, the script will do: nothing at install time (your first goal reached 🎉) and at first run generates certificates in ~/.https-localhost.

At this point your problem at install time will be solved, but is moved at run time. Furthermore, if you run https-localhost on linux you will be asked for sudo password also for the 443 and 80 ports (#11).

The work-around for the latter is easy: if you are able to change the port, using a highest number (like 4443 and 8080) the problem is solved. I'll add a CLI options for you to specify the port (#12).

For the first problem (certificates generation) I need further informations:
1. You are running under root on Jenkins? If not, is a problem to run the script under root?
2. Do you need the certificates to be trusted (the green lock in the image)? Or you just need to serve your content over https, and doesn't matter if the certificate is valid or not?

![green-lock](https://user-images.githubusercontent.com/1225294/51066373-96d4aa80-15be-11e9-91e2-f4e44a3a4458.png)

**Comment by ejunker** [comment]
Thanks for your reply. Yes, I think as long as an `npm install` won't cause it to ask for a password my issue will be solved. I don't need to use the certificates for anything that I am doing with Jenkins. I just need to npm install and build the app. Other Jenkins users might need to use the certificates and I would assume that they are not running under root. I think it would be nice if the certificates were trusted so that it works without having to disable certificate checking.

**Comment by daquinoaldo** [comment]
I'm working on that, I think it will be ready for the end of the next week.

Is not possible to trust certificates without root permissions, but I'll add an option to use a custom certificate, so that the certificates generation will be skipped.

**Comment by daquinoaldo** [comment]
Commit 05e7651 should solve the problem. I'll merge into the master branch this weekend, after some manual testing.

**Comment by daquinoaldo** [comment]
Should be solved with #13. Just upgrade your dependency to v4.1.0. Let me know!

**Comment by daquinoaldo** [comment]
Closed. Feel free to open again if you have any problem with the proposed solution.
